app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -2,42 +2,44 @@ class ManageIQ::Providers::Vmware::InfraManager::Inventory::Collector
   include PropertyCollector
   include Vmdb::Logging
 
-  def initialize(ems)
+  def initialize(ems, run_once: false)
     @ems             = ems
-    @exit_requested  = false
     @inventory_cache = ems.class::Inventory::Cache.new
+    @run_once        = run_once
+
+    self.exit_requested = false
   end
 
   def run
     until exit_requested
-      vim = connect(ems.hostname, ems.authentication_userid, ems.authentication_password)
-
-      begin
-        wait_for_updates(vim)
-      rescue RbVmomi::Fault => err
-        _log.error("Caught exception #{err.message}")
-        _log.log_backtrace(err)
-      ensure
-        vim.close unless vim.nil?
-        vim = nil
-      end
+      monitor_updates
+      break if run_once
     end
 
     _log.info("Exiting...")
-  ensure
-    vim.serviceContent.sessionManager.Logout unless vim.nil?
```

Comment:
and this? Are these remnants of the old broker?

Reply:
So this I replaced with https://github.com/ManageIQ/manageiq-providers-vmware/pull/234/files#diff-08ca91134d772c718409e3f5c83d8621R31 which does the same thing but just looks cleaner